apollo-router/src/services/router_service.rs

```diff
@@ -339,61 +324,208 @@ impl RouterService {
-                                            MULTIPART_DEFER_CONTENT_TYPE
-                                        ))
-                                        .extension_code("INVALID_ACCEPT_HEADER")
-                                        .build(),
-                                )
-                                .status_code(StatusCode::NOT_ACCEPTABLE)
-                                .header(CONTENT_TYPE, APPLICATION_JSON.essence_str())
-                                .context(context)
-                                .build()
+                    results.push(router::Response::error_builder()
+                            .error(
+                                graphql::Error::builder()
+                                    .message(format!(
+                                        r#"'accept' header must be one of: \"*/*\", {:?}, {:?} or {:?}"#,
+                                        APPLICATION_JSON.essence_str(),
+                                        GRAPHQL_JSON_RESPONSE_HEADER_VALUE,
+                                        MULTIPART_DEFER_CONTENT_TYPE
+                                    ))
+                                    .extension_code("INVALID_ACCEPT_HEADER")
+                                    .build(),
+                            )
+                            .status_code(StatusCode::NOT_ACCEPTABLE)
+                            .header(CONTENT_TYPE, APPLICATION_JSON.essence_str())
+                            .context(context)
+                            .build()?);
                 }
             }
         }
+        Ok(())
+    }
+
+    async fn call_inner(&self, req: RouterRequest) -> Result<RouterResponse, BoxError> {
+        let context = req.context.clone();
+
+        let supergraph_requests = match self.translate_request(req).await {
+            Ok(requests) => requests,
+            Err(err) => {
+                ::tracing::error!(
+                    monotonic_counter.apollo_router_http_requests_total = 1u64,
+                    status = %err.status.as_u16(),
+                    error = %err.error,
+                );
+
+                return router::Response::error_builder()
+                    .error(
+                        graphql::Error::builder()
+                            .message(String::from("Invalid GraphQL request"))
+                            .extension_code(err.extension_code)
+                            .extension("details", err.extension_details)
+                            .build(),
+                    )
+                    .status_code(err.status)
+                    .header(CONTENT_TYPE, APPLICATION_JSON.essence_str())
+                    .context(context)
+                    .build();
+            }
+        };
+
+        let mut results = vec![];
+
+        for supergraph_request in supergraph_requests {
+            self.process_supergraph_request(&mut results, supergraph_request)
+                .await?;
+        }
```

Comment:
Can we predict a minimal capacity using the `supergraph_requests.len()` ?

Reply:
done: 6088b38a8b78078c2a8dba66e1d11e64b57783fd